methods/basic.go

```diff
@@ -15,7 +20,62 @@ func BasicRequestWithBody(c *cli.Context, method string) (string, error) {
 		return "", err
 	}
 
-	var jsonStr = []byte(c.String("body"))
+	var jsonStr []byte
+
+	if c.Bool("editor") {
+		var path string
+		var editor string
+
+		// check OS
+		currentOs := runtime.GOOS
+
+		// based on OS find assign default editor
+		if currentOs == "linux" || currentOs == "darwin" {
+			editor = "nano"
+		} else if currentOs == "windows" {
+			editor = "notepad"
+		}
+
+		path, err = exec.LookPath(editor)
+		if err != nil {
+			return "", fmt.Errorf("Unable to locate %s at %s", editor, path)
```

Comment:
Generally, when bubbling up errors, it is a good idea to wrap the original (Errorf format verb `%w` in go1.13+) so you get the exact reason for the failure. Exceptions exist, but since this is running on a local user's system as a cli tool, this is relevant info.

Also, if err is not nil, path will be an empty string, so including the value of `path` will just be include an empty string.

Eg:
```go
return "", fmt.Errorf("unable to locate %s: %w", editor, err)
```